bindings/dart/lib/src/common/reranker/data_provider.dart

```diff
@@ -57,9 +57,31 @@ class Checksum {
 }
 
 /// The optional features to be enabled by picking platform dependent assets.
-enum Feature {
-  /// Web multi-threading features.
-  webParallel,
+class Feature {
+  // The constructor is private and is only callable within this file.
+  // We use the constructor only for creating the unique `WebFeature`/`WasmFeature`s.
+  Feature._();
+}
+
+class MobileFeature extends Feature {
+  MobileFeature._() : super._() {
+    // make sure no instance of this class can exists
+    throw UnimplementedError('This class cannot be instantiated');
+  }
+}
+
+class WebFeature extends Feature {
+  WebFeature._() : super._();
+}
+
+// When we compare `WasmFeature`s with each other, we rely on their object identity.
+// To avoid unexpected results when comparing them, we have to make sure that a
+// `WebFeature` cannot be created by a user but that the user only uses
+// `WasmFeature`.
+class WasmFeature {
```

Comment:
I find having both `WebFeature` and `WasmFeature` a bit confusing. Moreover this are all javascript features and are not related to wasm. If the problem was the name we can move directly the static fields in `WebFeature` and have `WebFeature.bulkMemory`. wdyt?

Reply:
sounds good 👍